src/sys/socket/mod.rs

```diff
@@ -677,6 +698,52 @@ pub enum ControlMessage<'a> {
     #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
     ScmCreds,
 
+    /// A message of type `IP_ORIGDSTADDR` which is triggered by setting
+    /// `RecvOrigDstAddrV4` socket option and is used to get original IPv4
+    /// UDP or TCP destination address when actual destination address
+    /// is overwritten by [`TPROXY`][1], which in turn may be used to
+    /// make a transparent proxy server.
+    /// 
+    /// [1]:https://www.kernel.org/doc/Documentation/networking/tproxy.txt
+    #[cfg(any(
+        target_os = "linux",
+        target_os = "android",
+    ))]
+    OrigDstAddrV4(&'a libc::sockaddr_in),
```

Comment:
There is no point in defining `ControlMessage::OrigDstAddrV[46]`.  `ControlMessage` is for use with `sendmsg` only.  `recvmsg` uses `ControlMessageOwned` instead.